test/unit/specs/vislib/lib/validate_visdata.js

```diff
@@ -0,0 +1,114 @@
+define(function (require) {
+  var angular = require('angular');
+  var _ = require('lodash');
+  var $ = require('jquery');
+
+  angular.module('ValidateVisDataFactory', ['kibana']);
+
+  describe('Vislib ValidateVisData Test Suite', function () {
+    function generateData(length, size, type) {
+      function generateValues(size, type) {
+        return Array.apply(null, new Array(size))
+        .map(function () {
+          if (type === 'pie') {
+            return {
+              size: +(Math.random() * 100).toFixed(0)
+            };
+          }
+
+          return {
+            x: +(Math.random() * 100).toFixed(0),
+            y: +(Math.random() * 100).toFixed(0)
+          };
+        });
+      }
+
+      return {
+        rows: Array.apply(null, new Array(length))
+        .map(function () {
+          if (type === 'pie') {
+            return {
+              slices: {
+                children: generateValues(size, type)
+              }
+            };
+          }
+
+          return { series: generateValues(size) };
+        })
+      };
+    }
+
+    // Add emp ty array
+    function addNullValues(data) {
+      _.forIn(data, function (value, key) {
+        if (key === 'pie') { value.rows.push({slices: {children: []}}); }
+        else {
+          value.rows.push({series: []});
+        }
+      });
+
+      ++length; // increment length counter
+      return data;
+    }
+
+    var length = 2;
+    var size = 10;
+    var data = {
+      pie: generateData(length, size, 'pie'),
+      series: generateData(length, size)
+    };
+    var ValidateVisData;
+    var validData;
+
+    addNullValues(data);
+
+    beforeEach(function () {
+      module('ValidateVisDataFactory');
+      inject(function (Private) {
+        ValidateVisData = Private(require('components/vislib/lib/validate_visdata'));
+        validData = new ValidateVisData(data.series);
+      });
+    });
+
+    afterEach(function () {
+      validData = null;
+    });
+
+    describe('_validate method', function () {
+      it('should remove null arrays from chart objects', function () {
+        var validSeriesData = validData._validate(data.series, 'histogram');
+        var validPieData = validData._validate(data.pie, 'pie');
```

Comment:
The `validData` object used here was created with `data.series`, but is getting `data.pie` passed to `_validate`